src/solvers/refinement/string_constraint_generator_main.cpp

```diff
@@ -236,9 +237,13 @@ string_exprt string_constraint_generatort::add_axioms_for_refined_string(
   }
   else
   {
-    throw "add_axioms_for_refined_string:\n"+string.pretty()+
-      "\nwhich is not a function application, "+
-      "a symbol, a struct or an if expression";
+    INVARIANT(
+      false,
```

Comment:
`INVARIANT(false, reason)` is kind of ugly, I wonder whether we should have `UNREACHABLE(reason)` for these situations.